packages/gitbook/src/components/Search/server-actions.tsx

```diff
@@ -60,6 +60,10 @@ export async function searchSiteContent(args: {
     const { siteSpaceIds, query, cacheBust } = args;
     const pointer = getContentPointer();
 
+    if (query.length < 3) {
+        return [];
+    }
```

Comment:
`C#` 😅

Reply:
Or lexigraphs could be an issue.

Wouldn't the increase in debounce time be an improvement already?

Reply:
yes, both would be an improvement really. a search like C# or a 2 char lexigraph could be argued that it wouldn't produce any useful results most of the time but I see your point. a higher debounce would help even more. At least, maybe we can agree that single char searches are not useful then and we can reduce that to 2. We are chipping away at the effectiveness of the improvement though.

Reply:
switched it to one for now. The debounce increase will improve it a bit already so let's see how much over the next few days.